{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "90ab4983",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.append(\"/Users/liangxuntan/Code/fyp2025/scripts\")\n",
    "\n",
    "import gmcfunctions as gmc\n",
    "import mlfunctions as mlf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4cf9d979",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error 404 for 4\n",
      "Error 404 for 5\n",
      "Error 404 for 15\n",
      "Error 404 for 16\n",
      "Error 404 for 17\n",
      "Error 404 for 18\n",
      "Error 404 for 19\n",
      "Error 404 for 20\n",
      "Error 404 for 22\n",
      "Error 404 for 23\n",
      "Error 404 for 24\n",
      "Error 404 for 30\n",
      "Error 404 for 32\n",
      "Error 404 for 33\n",
      "Error 404 for 41\n",
      "Error 404 for 42\n",
      "Error 404 for 43\n",
      "Error 404 for 44\n",
      "Error 404 for 45\n",
      "Error 404 for 46\n",
      "Error 404 for 47\n",
      "Error 404 for 50\n",
      "Error 404 for 62\n",
      "Error 404 for 64\n",
      "Error 404 for 66\n",
      "Error 404 for 72\n",
      "Error 404 for 89\n",
      "Error 404 for 94\n",
      "Error 404 for 96\n",
      "Error 404 for 99\n",
      "Error 404 for 100\n",
      "Error 404 for 102\n",
      "Error 404 for 103\n",
      "Error 404 for 106\n",
      "Error 404 for 107\n",
      "Error 404 for 112\n",
      "Error 404 for 116\n",
      "Error 404 for 117\n",
      "Error 404 for 128\n",
      "Error 404 for 131\n",
      "Error 404 for 132\n",
      "Error 404 for 145\n",
      "Error 404 for 151\n",
      "Error 404 for 154\n",
      "Error 404 for 164\n",
      "Error 404 for 168\n",
      "Error 404 for 175\n",
      "Error 404 for 176\n",
      "Error 404 for 177\n",
      "Error 404 for 178\n",
      "Error 404 for 179\n",
      "Error 404 for 180\n",
      "Error 404 for 184\n",
      "No valid 'bacdive_id' found for 187\n",
      "No valid 'bacdive_id' found for 188\n",
      "No valid 'bacdive_id' found for 190\n",
      "No valid 'bacdive_id' found for 191\n",
      "Error 404 for 197\n",
      "Error 404 for 199\n",
      "{1: 654, 2: 11979, 3: 12751, 6: 665, 7: 10435, 8: 14385, 9: 10177, 10: 14289, 11: 24124, 12: 824, 13: 413, 14: 2587, 21: 2507, 25: 59, 26: 1798, 27: 14008, 28: 2436, 29: 2403, 31: 572, 34: 2288, 35: 121, 36: 6145, 37: 10435, 38: 2717, 39: 2476, 40: 2401, 48: 2551, 49: 5513, 51: 17939, 52: 2606, 53: 10852, 54: 17579, 55: 2005, 56: 169541, 57: 6639, 58: 1723, 59: 6810, 60: 16811, 61: 16824, 63: 4085, 65: 10957, 67: 5514, 68: 6088, 69: 17415, 70: 111, 71: 112, 73: 833, 74: 16717, 75: 8068, 76: 2537, 77: 2660, 78: 2599, 79: 17077, 80: 13355, 81: 2965, 82: 14754, 83: 13414, 84: 16436, 85: 8215, 86: 16714, 87: 2417, 88: 16649, 90: 5707, 91: 12644, 92: 14445, 93: 6530, 95: 4226, 97: 5908, 98: 13623, 101: 12685, 104: 17860, 105: 1, 108: 30, 109: 6158, 110: 2725, 111: 1663, 113: 6141, 114: 6014, 115: 17265, 118: 723, 119: 6957, 120: 7087, 121: 2227, 122: 2729, 123: 1565, 124: 11789, 125: 7150, 126: 8284, 127: 7969, 129: 130852, 130: 13362, 133: 12810, 134: 14044, 135: 5402, 136: 17168, 137: 14300, 138: 2705, 139: 2600, 140: 2567, 141: 6993, 142: 11914, 143: 2737, 144: 16834, 146: 5917, 147: 16635, 148: 4229, 149: 4083, 150: 16640, 152: 5414, 153: 17150, 155: 17076, 156: 2646, 157: 6116, 158: 17017, 159: 14340, 160: 3767, 161: 158335, 162: 6180, 163: 4074, 165: 17674, 166: 6184, 167: 3826, 169: 14317, 170: 17477, 171: 16823, 172: 5497, 173: 14319, 174: 14320, 181: 17156, 182: 16650, 183: 2281, 185: 17031, 186: 7572, 189: 7813, 192: 174657, 193: 4106, 194: 4004, 195: 3959, 196: 4019, 198: 3993}\n",
      "140\n"
     ]
    }
   ],
   "source": [
    "medianumlist= range(1, 200)\n",
    "medianum_bacid_dict=gmc.get_first_bacdive_ids2(medianumlist,remove_none=True)\n",
    "print(medianum_bacid_dict)\n",
    "print(len(medianum_bacid_dict))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e0885d18",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Authentication successful --\n",
      "medium 1: BacDive ID → 654\n",
      "  » species-level taxid 1398\n",
      "medium 2: BacDive ID → 11979\n",
      "  » species-level taxid 1474\n",
      "medium 3: BacDive ID → 12751\n",
      "  » species-level taxid 354\n",
      "medium 6: BacDive ID → 665\n",
      "  » species-level taxid 1458\n",
      "medium 7: BacDive ID → 10435\n",
      "  » species-level taxid 190\n",
      "medium 8: BacDive ID → 14385\n",
      "  » species-level taxid 33018\n",
      "medium 9: BacDive ID → 10177\n",
      "  » species-level taxid 54\n",
      "medium 10: BacDive ID → 14289\n",
      "  » species-level taxid 542\n",
      "medium 11: BacDive ID → 24124\n",
      "medium 12: BacDive ID → 824\n",
      "  » species-level taxid 83428\n",
      "medium 13: BacDive ID → 413\n",
      "  » species-level taxid 122960\n",
      "medium 14: BacDive ID → 2587\n",
      "  » species-level taxid 1497\n",
      "medium 21: BacDive ID → 2507\n",
      "  » species-level taxid 1267\n",
      "medium 25: BacDive ID → 59\n",
      "  » species-level taxid 1071\n",
      "medium 26: BacDive ID → 1798\n",
      "  » species-level taxid 1074\n",
      "medium 27: BacDive ID → 14008\n",
      "  » species-level taxid 1085\n",
      "medium 28: BacDive ID → 2436\n",
      "  » species-level taxid 61595\n",
      "medium 29: BacDive ID → 2403\n",
      "  » species-level taxid 1092\n",
      "medium 31: BacDive ID → 572\n",
      "  » species-level taxid 1445\n",
      "medium 34: BacDive ID → 2288\n",
      "  » species-level taxid 33977\n",
      "medium 35: BacDive ID → 121\n",
      "  » species-level taxid 930\n",
      "medium 36: BacDive ID → 6145\n",
      "  » species-level taxid 931\n",
      "medium 37: BacDive ID → 10435\n",
      "  » species-level taxid 190\n",
      "medium 38: BacDive ID → 2717\n",
      "  » species-level taxid 1511\n",
      "medium 39: BacDive ID → 2476\n",
      "  » species-level taxid 12934\n",
      "medium 40: BacDive ID → 2401\n",
      "  » species-level taxid 1098\n",
      "medium 48: BacDive ID → 2551\n",
      "  » species-level taxid 1520\n",
      "medium 49: BacDive ID → 5513\n",
      "  » species-level taxid 983\n",
      "medium 51: BacDive ID → 17939\n",
      "  » species-level taxid 522889\n",
      "medium 52: BacDive ID → 2606\n",
      "  » species-level taxid 1534\n",
      "medium 53: BacDive ID → 10852\n",
      "  » species-level taxid 1833\n",
      "medium 54: BacDive ID → 17579\n",
      "  » species-level taxid 339\n",
      "medium 55: BacDive ID → 2005\n",
      "  » species-level taxid 106590\n",
      "medium 56: BacDive ID → 169541\n",
      "  » species-level taxid 33915\n",
      "medium 57: BacDive ID → 6639\n",
      "  » species-level taxid 238013\n",
      "medium 58: BacDive ID → 1723\n",
      "  » species-level taxid 78346\n",
      "medium 59: BacDive ID → 6810\n",
      "  » species-level taxid 1247\n",
      "medium 60: BacDive ID → 16811\n",
      "  » species-level taxid 1525\n",
      "medium 61: BacDive ID → 16824\n",
      "  » species-level taxid 1516\n",
      "medium 63: BacDive ID → 4085\n",
      "  » species-level taxid 885\n",
      "medium 65: BacDive ID → 10957\n",
      "  » species-level taxid 194249\n",
      "medium 67: BacDive ID → 5514\n",
      "  » species-level taxid 1337839\n",
      "medium 68: BacDive ID → 6088\n",
      "  » species-level taxid 927\n",
      "medium 69: BacDive ID → 17415\n",
      "  » species-level taxid 921\n",
      "medium 70: BacDive ID → 111\n",
      "  » species-level taxid 920\n",
      "medium 71: BacDive ID → 112\n",
      "  » species-level taxid 920\n",
      "medium 73: BacDive ID → 833\n",
      "  » species-level taxid 83428\n",
      "medium 74: BacDive ID → 16717\n",
      "  » species-level taxid 274\n",
      "medium 75: BacDive ID → 8068\n",
      "  » species-level taxid 68992\n",
      "medium 76: BacDive ID → 2537\n",
      "  » species-level taxid 1556\n",
      "medium 77: BacDive ID → 2660\n",
      "  » species-level taxid 29369\n",
      "medium 78: BacDive ID → 2599\n",
      "  » species-level taxid 1498\n",
      "medium 79: BacDive ID → 17077\n",
      "  » species-level taxid 45248\n",
      "medium 80: BacDive ID → 13355\n",
      "  » species-level taxid 37331\n",
      "medium 81: BacDive ID → 2965\n",
      "  » species-level taxid 65655\n",
      "medium 82: BacDive ID → 14754\n",
      "  » species-level taxid 1314\n",
      "medium 83: BacDive ID → 13414\n",
      "  » species-level taxid 1852\n",
      "medium 84: BacDive ID → 16436\n",
      "medium 85: BacDive ID → 8215\n",
      "  » species-level taxid 1792\n",
      "medium 86: BacDive ID → 16714\n",
      "  » species-level taxid 271\n",
      "medium 87: BacDive ID → 2417\n",
      "  » species-level taxid 1108\n",
      "medium 88: BacDive ID → 16649\n",
      "  » species-level taxid 2285\n",
      "medium 90: BacDive ID → 5707\n",
      "  » species-level taxid 510955\n",
      "medium 91: BacDive ID → 12644\n",
      "  » species-level taxid 1744\n",
      "medium 92: BacDive ID → 14445\n",
      "  » species-level taxid 1280\n",
      "medium 93: BacDive ID → 6530\n",
      "  » species-level taxid 152335\n",
      "medium 95: BacDive ID → 4226\n",
      "  » species-level taxid 891\n",
      "medium 97: BacDive ID → 5908\n",
      "  » species-level taxid 2242\n",
      "medium 98: BacDive ID → 13623\n",
      "  » species-level taxid 34013\n",
      "medium 101: BacDive ID → 12685\n",
      "  » species-level taxid 28109\n",
      "medium 104: BacDive ID → 17860\n",
      "  » species-level taxid 40542\n",
      "medium 105: BacDive ID → 1\n",
      "  » species-level taxid 435\n",
      "medium 108: BacDive ID → 30\n",
      "  » species-level taxid 76588\n",
      "medium 109: BacDive ID → 6158\n",
      "  » species-level taxid 50712\n",
      "medium 110: BacDive ID → 2725\n",
      "  » species-level taxid 1553\n",
      "medium 111: BacDive ID → 1663\n",
      "  » species-level taxid 533\n",
      "medium 113: BacDive ID → 6141\n",
      "  » species-level taxid 36861\n",
      "medium 114: BacDive ID → 6014\n",
      "  » species-level taxid 28252\n",
      "medium 115: BacDive ID → 17265\n",
      "  » species-level taxid 691\n",
      "medium 118: BacDive ID → 723\n",
      "  » species-level taxid 1404\n",
      "medium 119: BacDive ID → 6957\n",
      "  » species-level taxid 39441\n",
      "medium 120: BacDive ID → 7087\n",
      "  » species-level taxid 2208\n",
      "medium 121: BacDive ID → 2227\n",
      "  » species-level taxid 2053622\n",
      "medium 122: BacDive ID → 2729\n",
      "  » species-level taxid 1515\n",
      "medium 123: BacDive ID → 1565\n",
      "  » species-level taxid 1461\n",
      "medium 124: BacDive ID → 11789\n",
      "  » species-level taxid 58138\n",
      "medium 125: BacDive ID → 7150\n",
      "  » species-level taxid 410\n",
      "medium 126: BacDive ID → 8284\n",
      "  » species-level taxid 1771\n",
      "medium 127: BacDive ID → 7969\n",
      "  » species-level taxid 1877\n",
      "medium 129: BacDive ID → 130852\n",
      "  » species-level taxid 682598\n",
      "medium 130: BacDive ID → 13362\n",
      "  » species-level taxid 33911\n",
      "medium 133: BacDive ID → 12810\n",
      "  » species-level taxid 290\n",
      "medium 134: BacDive ID → 14044\n",
      "  » species-level taxid 2350\n",
      "medium 135: BacDive ID → 5402\n",
      "  » species-level taxid 33952\n",
      "medium 136: BacDive ID → 17168\n",
      "  » species-level taxid 39777\n",
      "medium 137: BacDive ID → 14300\n",
      "  » species-level taxid 1890675\n",
      "medium 138: BacDive ID → 2705\n",
      "  » species-level taxid 29370\n",
      "medium 139: BacDive ID → 2600\n",
      "  » species-level taxid 1498\n",
      "medium 140: BacDive ID → 2567\n",
      "  » species-level taxid 29355\n",
      "medium 141: BacDive ID → 6993\n",
      "  » species-level taxid 2187\n",
      "medium 142: BacDive ID → 11914\n",
      "  » species-level taxid 934\n",
      "medium 143: BacDive ID → 2737\n",
      "  » species-level taxid 1519\n",
      "medium 144: BacDive ID → 16834\n",
      "  » species-level taxid 29323\n",
      "medium 146: BacDive ID → 5917\n",
      "  » species-level taxid 2250\n",
      "medium 147: BacDive ID → 16635\n",
      "  » species-level taxid 867\n",
      "medium 148: BacDive ID → 4229\n",
      "  » species-level taxid 38815\n",
      "medium 149: BacDive ID → 4083\n",
      "  » species-level taxid 879\n",
      "medium 150: BacDive ID → 16640\n",
      "  » species-level taxid 41673\n",
      "medium 152: BacDive ID → 5414\n",
      "  » species-level taxid 1528\n",
      "medium 153: BacDive ID → 17150\n",
      "  » species-level taxid 209080\n",
      "medium 155: BacDive ID → 17076\n",
      "  » species-level taxid 1022\n",
      "medium 156: BacDive ID → 2646\n",
      "  » species-level taxid 28446\n",
      "medium 157: BacDive ID → 6116\n",
      "  » species-level taxid 844\n",
      "medium 158: BacDive ID → 17017\n",
      "  » species-level taxid 2303\n",
      "medium 159: BacDive ID → 14340\n",
      "  » species-level taxid 163\n",
      "medium 160: BacDive ID → 3767\n",
      "  » species-level taxid 985\n",
      "medium 161: BacDive ID → 158335\n",
      "  » species-level taxid 2052835\n",
      "medium 162: BacDive ID → 6180\n",
      "  » species-level taxid 53398\n",
      "medium 163: BacDive ID → 4074\n",
      "  » species-level taxid 876\n",
      "medium 165: BacDive ID → 17674\n",
      "  » species-level taxid 35830\n",
      "medium 166: BacDive ID → 6184\n",
      "  » species-level taxid 53399\n",
      "medium 167: BacDive ID → 3826\n",
      "  » species-level taxid 153721\n",
      "medium 169: BacDive ID → 14317\n",
      "  » species-level taxid 147\n",
      "medium 170: BacDive ID → 17477\n",
      "  » species-level taxid 262324\n",
      "medium 171: BacDive ID → 16823\n",
      "  » species-level taxid 2325\n",
      "medium 172: BacDive ID → 5497\n",
      "  » species-level taxid 979\n",
      "medium 173: BacDive ID → 14319\n",
      "  » species-level taxid 151\n",
      "medium 174: BacDive ID → 14320\n",
      "  » species-level taxid 152\n",
      "medium 181: BacDive ID → 17156\n",
      "  » species-level taxid 971\n",
      "medium 182: BacDive ID → 16650\n",
      "  » species-level taxid 2287\n",
      "medium 183: BacDive ID → 2281\n",
      "  » species-level taxid 82803\n",
      "medium 185: BacDive ID → 17031\n",
      "  » species-level taxid 2271\n",
      "medium 186: BacDive ID → 7572\n",
      "  » species-level taxid 1673\n",
      "medium 189: BacDive ID → 7813\n",
      "  » species-level taxid 52697\n",
      "medium 192: BacDive ID → 174657\n",
      "  » species-level taxid 1313\n",
      "medium 193: BacDive ID → 4106\n",
      "  » species-level taxid 881\n",
      "medium 194: BacDive ID → 4004\n",
      "  » species-level taxid 894\n",
      "medium 195: BacDive ID → 3959\n",
      "  » species-level taxid 449304\n",
      "medium 196: BacDive ID → 4019\n",
      "  » species-level taxid 65555\n",
      "medium 198: BacDive ID → 3993\n",
      "  » species-level taxid 2300\n",
      "{1398: 1, 1474: 2, 354: 3, 1458: 6, 190: 37, 33018: 8, 54: 9, 542: 10, 83428: 73, 122960: 13, 1497: 14, 1267: 21, 1071: 25, 1074: 26, 1085: 27, 61595: 28, 1092: 29, 1445: 31, 33977: 34, 930: 35, 931: 36, 1511: 38, 12934: 39, 1098: 40, 1520: 48, 983: 49, 522889: 51, 1534: 52, 1833: 53, 339: 54, 106590: 55, 33915: 56, 238013: 57, 78346: 58, 1247: 59, 1525: 60, 1516: 61, 885: 63, 194249: 65, 1337839: 67, 927: 68, 921: 69, 920: 71, 274: 74, 68992: 75, 1556: 76, 29369: 77, 1498: 139, 45248: 79, 37331: 80, 65655: 81, 1314: 82, 1852: 83, 1792: 85, 271: 86, 1108: 87, 2285: 88, 510955: 90, 1744: 91, 1280: 92, 152335: 93, 891: 95, 2242: 97, 34013: 98, 28109: 101, 40542: 104, 435: 105, 76588: 108, 50712: 109, 1553: 110, 533: 111, 36861: 113, 28252: 114, 691: 115, 1404: 118, 39441: 119, 2208: 120, 2053622: 121, 1515: 122, 1461: 123, 58138: 124, 410: 125, 1771: 126, 1877: 127, 682598: 129, 33911: 130, 290: 133, 2350: 134, 33952: 135, 39777: 136, 1890675: 137, 29370: 138, 29355: 140, 2187: 141, 934: 142, 1519: 143, 29323: 144, 2250: 146, 867: 147, 38815: 148, 879: 149, 41673: 150, 1528: 152, 209080: 153, 1022: 155, 28446: 156, 844: 157, 2303: 158, 163: 159, 985: 160, 2052835: 161, 53398: 162, 876: 163, 35830: 165, 53399: 166, 153721: 167, 147: 169, 262324: 170, 2325: 171, 979: 172, 151: 173, 152: 174, 971: 181, 2287: 182, 82803: 183, 2271: 185, 1673: 186, 52697: 189, 1313: 192, 881: 193, 894: 194, 449304: 195, 65555: 196, 2300: 198}\n"
     ]
    }
   ],
   "source": [
    "taxid_medianum_dict=gmc.map_species_taxids_to_media(\n",
    "                    medianums=list(medianum_bacid_dict.keys()),\n",
    "                    bacdive_id_dict=medianum_bacid_dict,\n",
    "                    email=\"tanliangxun2000@gmail.com\",\n",
    "                    password=\"pass1234\",)\n",
    "print(taxid_medianum_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "41c688b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: /Users/liangxuntan/Code/fyp2025/scripts/run_taxid2fna.sh 1398 1474 354 1458 190 33018 54 542 83428 122960 1497 1267 1071 1074 1085 61595 1092 1445 33977 930 931 1511 12934 1098 1520 983 522889 1534 1833 339 106590 33915 238013 78346 1247 1525 1516 885 194249 1337839 927 921 920 274 68992 1556 29369 1498 45248 37331 65655 1314 1852 1792 271 1108 2285 510955 1744 1280 152335 891 2242 34013 28109 40542 435 76588 50712 1553 533 36861 28252 691 1404 39441 2208 2053622 1515 1461 58138 410 1771 1877 682598 33911 290 2350 33952 39777 1890675 29370 29355 2187 934 1519 29323 2250 867 38815 879 41673 1528 209080 1022 28446 844 2303 163 985 2052835 53398 876 35830 53399 153721 147 262324 2325 979 151 152 971 2287 82803 2271 1673 52697 1313 881 894 449304 65555 2300\n",
      "→ Processing TaxID 1398\n",
      "→ Processing TaxID 1474\n",
      "→ Processing TaxID 354\n",
      "→ Processing TaxID 1458\n",
      "→ Processing TaxID 190\n",
      "→ Processing TaxID 33018\n",
      "→ Processing TaxID 54\n",
      "→ Processing TaxID 542\n",
      "→ Processing TaxID 83428\n",
      "→ Processing TaxID 122960\n",
      "→ Processing TaxID 1497\n",
      "→ Processing TaxID 1267\n",
      "→ Processing TaxID 1071\n",
      "→ Processing TaxID 1074\n",
      "→ Processing TaxID 1085\n",
      "→ Processing TaxID 61595\n",
      "→ Processing TaxID 1092\n",
      "→ Processing TaxID 1445\n",
      "→ Processing TaxID 33977\n",
      "→ Processing TaxID 930\n",
      "→ Processing TaxID 931\n",
      "→ Processing TaxID 1511\n",
      "→ Processing TaxID 12934\n",
      "→ Processing TaxID 1098\n",
      "→ Processing TaxID 1520\n",
      "→ Processing TaxID 983\n",
      "→ Processing TaxID 522889\n",
      "→ Processing TaxID 1534\n",
      "→ Processing TaxID 1833\n",
      "→ Processing TaxID 339\n",
      "→ Processing TaxID 106590\n",
      "→ Processing TaxID 33915\n",
      "→ Processing TaxID 238013\n",
      "→ Processing TaxID 78346\n",
      "→ Processing TaxID 1247\n",
      "→ Processing TaxID 1525\n",
      "→ Processing TaxID 1516\n",
      "→ Processing TaxID 885\n",
      "→ Processing TaxID 194249\n",
      "→ Processing TaxID 1337839\n",
      "→ Processing TaxID 927\n",
      "→ Processing TaxID 921\n",
      "→ Processing TaxID 920\n",
      "→ Processing TaxID 274\n",
      "→ Processing TaxID 68992\n",
      "→ Processing TaxID 1556\n",
      "→ Processing TaxID 29369\n",
      "→ Processing TaxID 1498\n",
      "→ Processing TaxID 45248\n",
      "→ Processing TaxID 37331\n",
      "→ Processing TaxID 65655\n",
      "→ Processing TaxID 1314\n",
      "→ Processing TaxID 1852\n",
      "→ Processing TaxID 1792\n",
      "→ Processing TaxID 271\n",
      "→ Processing TaxID 1108\n",
      "→ Processing TaxID 2285\n",
      "→ Processing TaxID 510955\n",
      "→ Processing TaxID 1744\n",
      "→ Processing TaxID 1280\n",
      "→ Processing TaxID 152335\n",
      "→ Processing TaxID 891\n",
      "→ Processing TaxID 2242\n",
      "→ Processing TaxID 34013\n",
      "→ Processing TaxID 28109\n",
      "→ Processing TaxID 40542\n",
      "→ Processing TaxID 435\n",
      "→ Processing TaxID 76588\n",
      "→ Processing TaxID 50712\n",
      "→ Processing TaxID 1553\n",
      "→ Processing TaxID 533\n",
      "→ Processing TaxID 36861\n",
      "→ Processing TaxID 28252\n",
      "→ Processing TaxID 691\n",
      "→ Processing TaxID 1404\n",
      "→ Processing TaxID 39441\n",
      "→ Processing TaxID 2208\n",
      "→ Processing TaxID 2053622\n",
      "→ Processing TaxID 1515\n",
      "→ Processing TaxID 1461\n",
      "→ Processing TaxID 58138\n",
      "→ Processing TaxID 410\n",
      "→ Processing TaxID 1771\n",
      "→ Processing TaxID 1877\n",
      "→ Processing TaxID 682598\n",
      "→ Processing TaxID 33911\n",
      "→ Processing TaxID 290\n",
      "→ Processing TaxID 2350\n",
      "→ Processing TaxID 33952\n",
      "→ Processing TaxID 39777\n",
      "→ Processing TaxID 1890675\n",
      "→ Processing TaxID 29370\n",
      "→ Processing TaxID 29355\n",
      "→ Processing TaxID 2187\n",
      "→ Processing TaxID 934\n",
      "→ Processing TaxID 1519\n",
      "→ Processing TaxID 29323\n",
      "→ Processing TaxID 2250\n",
      "→ Processing TaxID 867\n",
      "→ Processing TaxID 38815\n",
      "→ Processing TaxID 879\n",
      "→ Processing TaxID 41673\n",
      "→ Processing TaxID 1528\n",
      "→ Processing TaxID 209080\n",
      "→ Processing TaxID 1022\n",
      "→ Processing TaxID 28446\n",
      "→ Processing TaxID 844\n",
      "→ Processing TaxID 2303\n",
      "→ Processing TaxID 163\n",
      "→ Processing TaxID 985\n",
      "→ Processing TaxID 2052835\n",
      "→ Processing TaxID 53398\n",
      "→ Processing TaxID 876\n",
      "→ Processing TaxID 35830\n",
      "→ Processing TaxID 53399\n",
      "→ Processing TaxID 153721\n",
      "→ Processing TaxID 147\n",
      "→ Processing TaxID 262324\n",
      "→ Processing TaxID 2325\n",
      "→ Processing TaxID 979\n",
      "→ Processing TaxID 151\n",
      "→ Processing TaxID 152\n",
      "→ Processing TaxID 971\n",
      "→ Processing TaxID 2287\n",
      "→ Processing TaxID 82803\n",
      "→ Processing TaxID 2271\n",
      "→ Processing TaxID 1673\n",
      "→ Processing TaxID 52697\n",
      "→ Processing TaxID 1313\n",
      "→ Processing TaxID 881\n",
      "→ Processing TaxID 894\n",
      "→ Processing TaxID 449304\n",
      "→ Processing TaxID 65555\n",
      "→ Processing TaxID 2300\n",
      "✓ Finished. Full log: \"/Users/liangxuntan/Code/fyp2025/data/logs/taxid2fna_20250520_210124.log\"\n",
      "✅ Success: 1398 1474 354 1458 190 33018 54 542 122960 1497 1267 1071 1074 1085 61595 1092 1445 33977 930 931 1511 1098 1520 983 522889 1534 1833 339 106590 238013 78346 1247 1525 927 921 920 274 1556 29369 1498 45248 37331 65655 1314 1852 1792 271 1108 2285 510955 1744 1280 152335 891 2242 34013 28109 40542 435 76588 50712 1553 533 36861 28252 691 1404 39441 2208 1515 1461 58138 410 1771 1877 290 2350 33952 39777 29370 29355 2187 934 1519 29323 2250 867 38815 879 41673 1528 209080 1022 28446 844 2303 163 985 876 35830 53399 153721 262324 2325 979 971 2287 82803 2271 1673 52697 1313 881 894 65555 2300\n",
      "❌ Failed: 83428 12934 33915 1516 885 194249 1337839 68992 2053622 682598 33911 1890675 2052835 53398 147 151 152 449304\n"
     ]
    }
   ],
   "source": [
    "taxids= list(taxid_medianum_dict.keys())\n",
    "\n",
    "gmc.run_taxid2fna(taxids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b14cf339",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 151 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1445_predictedproteins.prot'\n",
      "Written 157 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1519_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 83 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1313_predictedproteins.prot'\n",
      "Written 166 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 194 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1497_predictedproteins.prot'\n",
      "Written 66 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1314_predictedproteins.prot'\n",
      "Written 250 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1520_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 153 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1022_predictedproteins.prot'\n",
      "Written 132 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1498_predictedproteins.prot'\n",
      "Written 152 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1461_predictedproteins.prot'\n",
      "Written 91 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1528_predictedproteins.prot'\n",
      "Written 139 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1511_predictedproteins.prot'\n",
      "Written 59 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/152335_predictedproteins.prot'\n",
      "Written 159 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1534_predictedproteins.prot'\n",
      "Written 121 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 187 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1085_predictedproteins.prot'\n",
      "Written 132 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1556_predictedproteins.prot'\n",
      "Written 131 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1515_predictedproteins.prot'\n",
      "Written 70 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/122960_predictedproteins.prot'\n",
      "Written 58 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot'\n",
      "Written 111 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/163_predictedproteins.prot'\n",
      "Written 45 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1247_predictedproteins.prot'\n",
      "Written 141 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/153721_predictedproteins.prot'\n",
      "Written 99 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1525_predictedproteins.prot'\n",
      "Written 117 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1267_predictedproteins.prot'\n",
      "Written 229 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1553_predictedproteins.prot'\n",
      "Written 317 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1771_predictedproteins.prot'\n",
      "Written 70 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1092_predictedproteins.prot'\n",
      "Written 126 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1280_predictedproteins.prot'\n",
      "Written 188 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1404_predictedproteins.prot'\n",
      "Written 74 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2285_predictedproteins.prot'\n",
      "Written 355 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1877_predictedproteins.prot'\n",
      "Written 72 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2208_predictedproteins.prot'\n",
      "Written 72 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2287_predictedproteins.prot'\n",
      "Written 100 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2325_predictedproteins.prot'\n",
      "Written 388 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/106590_predictedproteins.prot'\n",
      "Written 308 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1792_predictedproteins.prot'\n",
      "Written 57 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1098_predictedproteins.prot'\n",
      "Written 107 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2242_predictedproteins.prot'\n",
      "Written 103 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot'\n",
      "Written 138 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2250_predictedproteins.prot'\n",
      "Written 341 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1071_predictedproteins.prot'\n",
      "Written 141 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/209080_predictedproteins.prot'\n",
      "Written 82 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2187_predictedproteins.prot'\n",
      "Written 83 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2271_predictedproteins.prot'\n",
      "Written 336 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1673_predictedproteins.prot'\n",
      "Written 128 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/271_predictedproteins.prot'\n",
      "Written 94 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28446_predictedproteins.prot'\n",
      "Written 231 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1074_predictedproteins.prot'\n",
      "Written 103 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/274_predictedproteins.prot'\n",
      "Written 124 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1744_predictedproteins.prot'\n",
      "Written 204 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29369_predictedproteins.prot'\n",
      "Written 199 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1833_predictedproteins.prot'\n",
      "Written 139 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28109_predictedproteins.prot'\n",
      "Written 117 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33977_predictedproteins.prot'\n",
      "Written 147 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1108_predictedproteins.prot'\n",
      "Written 212 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2350_predictedproteins.prot'\n",
      "Written 134 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/290_predictedproteins.prot'\n",
      "Written 89 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/238013_predictedproteins.prot'\n",
      "Written 182 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1852_predictedproteins.prot'\n",
      "Written 99 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39441_predictedproteins.prot'\n",
      "Written 195 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29370_predictedproteins.prot'\n",
      "Written 122 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29323_predictedproteins.prot'\n",
      "Written 179 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/35830_predictedproteins.prot'\n",
      "Written 231 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2300_predictedproteins.prot'\n",
      "Written 102 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39777_predictedproteins.prot'\n",
      "Written 166 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28252_predictedproteins.prot'\n",
      "Written 107 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/40542_predictedproteins.prot'\n",
      "Written 108 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/41673_predictedproteins.prot'\n",
      "Written 79 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2303_predictedproteins.prot'\n",
      "Written 69 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot'\n",
      "Written 55 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/510955_predictedproteins.prot'\n",
      "Written 210 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29355_predictedproteins.prot'\n",
      "Written 248 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/262324_predictedproteins.prot'\n",
      "Written 258 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/410_predictedproteins.prot'\n",
      "Written 204 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/339_predictedproteins.prot'\n",
      "Written 135 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33952_predictedproteins.prot'\n",
      "Written 140 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/36861_predictedproteins.prot'\n",
      "Written 77 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/867_predictedproteins.prot'\n",
      "Written 250 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/522889_predictedproteins.prot'\n",
      "Written 363 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/37331_predictedproteins.prot'\n",
      "Written 96 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/435_predictedproteins.prot'\n",
      "Written 85 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/61595_predictedproteins.prot'\n",
      "Written 193 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/76588_predictedproteins.prot'\n",
      "Written 467 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/52697_predictedproteins.prot'\n",
      "Written 241 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/34013_predictedproteins.prot'\n",
      "Written 130 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/38815_predictedproteins.prot'\n",
      "Written 85 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65555_predictedproteins.prot'\n",
      "Written 104 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/533_predictedproteins.prot'\n",
      "Written 103 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/78346_predictedproteins.prot'\n",
      "Written 164 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/45248_predictedproteins.prot'\n",
      "Written 107 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/876_predictedproteins.prot'\n",
      "Written 70 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/542_predictedproteins.prot'\n",
      "Written 164 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/50712_predictedproteins.prot'\n",
      "Written 107 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/53399_predictedproteins.prot'\n",
      "Written 277 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65655_predictedproteins.prot'\n",
      "Written 276 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot'\n",
      "Written 139 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/891_predictedproteins.prot'\n",
      "Written 79 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/82803_predictedproteins.prot'\n",
      "Written 493 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot'\n",
      "Written 124 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/971_predictedproteins.prot'\n",
      "Written 47 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/844_predictedproteins.prot'\n",
      "Written 154 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/58138_predictedproteins.prot'\n",
      "Written 140 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/979_predictedproteins.prot'\n",
      "Written 160 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/879_predictedproteins.prot'\n",
      "Written 150 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/691_predictedproteins.prot'\n",
      "Written 65 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/927_predictedproteins.prot'\n",
      "Written 154 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/983_predictedproteins.prot'\n",
      "Written 95 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/985_predictedproteins.prot'\n",
      "Written 132 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/894_predictedproteins.prot'\n",
      "Written 145 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/881_predictedproteins.prot'\n",
      "Written 75 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/930_predictedproteins.prot'\n",
      "Written 93 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/920_predictedproteins.prot'\n",
      "Written 136 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/931_predictedproteins.prot'\n",
      "Written 256 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/921_predictedproteins.prot'\n",
      "Written 79 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/934_predictedproteins.prot'\n",
      "Protein prediction output files:\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1022_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/106590_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1071_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1074_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1085_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1092_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1098_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1108_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/122960_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1247_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1267_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1280_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1313_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1314_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1404_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1445_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1461_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1497_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1498_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1511_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1515_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1519_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1520_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/152335_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1525_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1528_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1534_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/153721_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1553_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1556_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/163_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1673_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1744_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1771_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1792_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1833_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1852_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1877_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/209080_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2187_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2208_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2242_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2250_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2271_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2285_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2287_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2300_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2303_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2325_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2350_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/238013_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/262324_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/271_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/274_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28109_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28252_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28446_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/290_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29323_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29355_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29369_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29370_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33952_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33977_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/339_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/34013_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/35830_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/36861_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/37331_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/38815_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39441_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39777_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/40542_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/410_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/41673_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/435_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/45248_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/50712_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/510955_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/522889_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/52697_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/53399_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/533_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/542_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/58138_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/61595_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65555_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65655_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/691_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/76588_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/78346_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/82803_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/844_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/867_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/876_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/879_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/881_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/891_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/894_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/920_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/921_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/927_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/930_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/931_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/934_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/971_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/979_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/983_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/985_predictedproteins.prot\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1022_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/106590_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1071_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1074_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1085_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1092_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1098_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1108_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/122960_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1247_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1267_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1280_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1313_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1314_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1404_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1445_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1461_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1497_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1498_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1511_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1515_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1519_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1520_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/152335_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1525_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1528_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1534_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/153721_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1553_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1556_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/163_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1673_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1744_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1771_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1792_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1833_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1852_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1877_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/209080_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2187_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2208_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2242_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2250_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2271_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2285_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2287_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2300_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2303_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2325_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2350_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/238013_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/262324_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/271_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/274_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28109_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28252_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28446_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/290_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29323_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29355_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29369_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29370_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33952_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33977_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/339_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/34013_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/35830_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/36861_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/37331_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/38815_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39441_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39777_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/40542_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/410_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/41673_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/435_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/45248_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/50712_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/510955_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/522889_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/52697_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/53399_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/533_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/542_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/58138_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/61595_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65555_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65655_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/691_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/76588_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/78346_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/82803_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/844_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/867_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/876_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/879_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/881_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/891_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/894_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/920_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/921_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/927_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/930_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/931_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/934_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/971_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/979_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/983_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/985_predictedproteins.prot')]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gmc.run_protein_prediction(\n",
    "    fasta_dir_path=\"/Users/liangxuntan/Code/fyp2025/data/labelled_genomes\",\n",
    "    evalue='1e-20',\n",
    "    nproc=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e09f0675",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: /Users/liangxuntan/Code/fyp2025/scripts/prot2pfam.sh -j 8 /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1022_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/106590_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1071_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1074_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1085_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1092_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1098_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1108_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/122960_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1247_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1267_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1280_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1313_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1314_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1404_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1445_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1461_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1497_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1498_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1511_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1515_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1519_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1520_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/152335_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1525_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1528_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1534_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/153721_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1553_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1556_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/163_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1673_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1744_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1771_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1792_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1833_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1852_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1877_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/209080_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2187_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2208_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2242_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2250_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2271_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2285_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2287_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2300_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2303_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2325_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/2350_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/238013_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/262324_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/271_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/274_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/28109_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/28252_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/28446_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/290_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/29323_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/29355_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/29369_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/29370_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/33952_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/33977_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/339_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/34013_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/35830_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/36861_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/37331_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/38815_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/39441_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/39777_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/40542_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/410_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/41673_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/435_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/45248_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/50712_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/510955_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/522889_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/52697_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/53399_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/533_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/542_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/58138_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/61595_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/65555_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/65655_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/691_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/76588_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/78346_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/82803_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/844_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/867_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/876_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/879_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/881_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/891_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/894_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/920_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/921_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/927_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/930_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/931_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/934_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/971_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/979_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/983_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/985_predictedproteins.prot\n",
      "▶ Starting parallel hmmscan ( Tue May 20 21:27:17 BST 2025 ) on 116 file(s)\n",
      "Running 8 job(s) × 1 thread(s) each\n",
      "→ Processing 1098_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1098_pfam.tbl\n",
      "✓ Finished 1098_predictedproteins.prot in 48s\n",
      "→ Processing 1092_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1092_pfam.tbl\n",
      "✓ Finished 1092_predictedproteins.prot in 58s\n",
      "→ Processing 1247_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1247_pfam.tbl\n",
      "✓ Finished 1247_predictedproteins.prot in 37s\n",
      "→ Processing 122960_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/122960_pfam.tbl\n",
      "✓ Finished 122960_predictedproteins.prot in 56s\n",
      "→ Processing 1108_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1108_pfam.tbl\n",
      "✓ Finished 1108_predictedproteins.prot in 120s\n",
      "→ Processing 1022_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1022_pfam.tbl\n",
      "✓ Finished 1022_predictedproteins.prot in 125s\n",
      "→ Processing 1085_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1085_pfam.tbl\n",
      "✓ Finished 1085_predictedproteins.prot in 153s\n",
      "→ Processing 1314_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1314_pfam.tbl\n",
      "✓ Finished 1314_predictedproteins.prot in 54s\n",
      "→ Processing 1074_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1074_pfam.tbl\n",
      "✓ Finished 1074_predictedproteins.prot in 188s\n",
      "→ Processing 1313_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1313_pfam.tbl\n",
      "✓ Finished 1313_predictedproteins.prot in 68s\n",
      "→ Processing 1267_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1267_pfam.tbl\n",
      "✓ Finished 1267_predictedproteins.prot in 94s\n",
      "→ Processing 1398_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1398_pfam.tbl\n",
      "✓ Finished 1398_predictedproteins.prot in 47s\n",
      "→ Processing 1280_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1280_pfam.tbl\n",
      "✓ Finished 1280_predictedproteins.prot in 102s\n",
      "→ Processing 1071_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1071_pfam.tbl\n",
      "✓ Finished 1071_predictedproteins.prot in 275s\n",
      "→ Processing 1474_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1474_pfam.tbl\n",
      "✓ Finished 1474_predictedproteins.prot in 98s\n",
      "→ Processing 1445_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1445_pfam.tbl\n",
      "✓ Finished 1445_predictedproteins.prot in 122s\n",
      "→ Processing 1461_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1461_pfam.tbl\n",
      "✓ Finished 1461_predictedproteins.prot in 123s\n",
      "→ Processing 106590_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/106590_pfam.tbl\n",
      "✓ Finished 106590_predictedproteins.prot in 313s\n",
      "→ Processing 1458_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1458_pfam.tbl\n",
      "✓ Finished 1458_predictedproteins.prot in 134s\n",
      "→ Processing 1404_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1404_pfam.tbl\n",
      "✓ Finished 1404_predictedproteins.prot in 152s\n",
      "→ Processing 1497_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1497_pfam.tbl\n",
      "✓ Finished 1497_predictedproteins.prot in 157s\n",
      "→ Processing 152335_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/152335_pfam.tbl\n",
      "✓ Finished 152335_predictedproteins.prot in 48s\n",
      "→ Processing 1498_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1498_pfam.tbl\n",
      "✓ Finished 1498_predictedproteins.prot in 107s\n",
      "→ Processing 1511_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1511_pfam.tbl\n",
      "✓ Finished 1511_predictedproteins.prot in 112s\n",
      "→ Processing 1525_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1525_pfam.tbl\n",
      "✓ Finished 1525_predictedproteins.prot in 81s\n",
      "→ Processing 1515_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1515_pfam.tbl\n",
      "✓ Finished 1515_predictedproteins.prot in 106s\n",
      "→ Processing 1528_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1528_pfam.tbl\n",
      "✓ Finished 1528_predictedproteins.prot in 73s\n",
      "→ Processing 1519_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1519_pfam.tbl\n",
      "✓ Finished 1519_predictedproteins.prot in 127s\n",
      "→ Processing 153721_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/153721_pfam.tbl\n",
      "✓ Finished 153721_predictedproteins.prot in 114s\n",
      "→ Processing 1534_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1534_pfam.tbl\n",
      "✓ Finished 1534_predictedproteins.prot in 128s\n",
      "→ Processing 163_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/163_pfam.tbl\n",
      "✓ Finished 163_predictedproteins.prot in 90s\n",
      "→ Processing 1520_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1520_pfam.tbl\n",
      "✓ Finished 1520_predictedproteins.prot in 201s\n",
      "→ Processing 1556_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1556_pfam.tbl\n",
      "✓ Finished 1556_predictedproteins.prot in 107s\n",
      "→ Processing 1744_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1744_pfam.tbl\n",
      "✓ Finished 1744_predictedproteins.prot in 100s\n",
      "→ Processing 1553_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1553_pfam.tbl\n",
      "✓ Finished 1553_predictedproteins.prot in 182s\n",
      "→ Processing 190_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/190_pfam.tbl\n",
      "✓ Finished 190_predictedproteins.prot in 82s\n",
      "→ Processing 1852_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1852_pfam.tbl\n",
      "✓ Finished 1852_predictedproteins.prot in 146s\n",
      "→ Processing 1833_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1833_pfam.tbl\n",
      "✓ Finished 1833_predictedproteins.prot in 159s\n",
      "→ Processing 2187_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2187_pfam.tbl\n",
      "✓ Finished 2187_predictedproteins.prot in 67s\n",
      "→ Processing 1673_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1673_pfam.tbl\n",
      "✓ Finished 1673_predictedproteins.prot in 267s\n",
      "→ Processing 209080_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/209080_pfam.tbl\n",
      "✓ Finished 209080_predictedproteins.prot in 115s\n",
      "→ Processing 2208_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2208_pfam.tbl\n",
      "✓ Finished 2208_predictedproteins.prot in 60s\n",
      "→ Processing 1792_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1792_pfam.tbl\n",
      "✓ Finished 1792_predictedproteins.prot in 248s\n",
      "→ Processing 1771_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1771_pfam.tbl\n",
      "✓ Finished 1771_predictedproteins.prot in 254s\n",
      "→ Processing 2242_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2242_pfam.tbl\n",
      "✓ Finished 2242_predictedproteins.prot in 87s\n",
      "→ Processing 2285_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2285_pfam.tbl\n",
      "✓ Finished 2285_predictedproteins.prot in 61s\n",
      "→ Processing 2271_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2271_pfam.tbl\n",
      "✓ Finished 2271_predictedproteins.prot in 68s\n",
      "→ Processing 2287_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2287_pfam.tbl\n",
      "✓ Finished 2287_predictedproteins.prot in 58s\n",
      "→ Processing 2250_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2250_pfam.tbl\n",
      "✓ Finished 2250_predictedproteins.prot in 112s\n",
      "→ Processing 1877_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1877_pfam.tbl\n",
      "✓ Finished 1877_predictedproteins.prot in 285s\n",
      "→ Processing 2303_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2303_pfam.tbl\n",
      "✓ Finished 2303_predictedproteins.prot in 63s\n",
      "→ Processing 2325_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2325_pfam.tbl\n",
      "✓ Finished 2325_predictedproteins.prot in 81s\n",
      "→ Processing 238013_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/238013_pfam.tbl\n",
      "✓ Finished 238013_predictedproteins.prot in 72s\n",
      "→ Processing 274_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/274_pfam.tbl\n",
      "✓ Finished 274_predictedproteins.prot in 83s\n",
      "→ Processing 271_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/271_pfam.tbl\n",
      "✓ Finished 271_predictedproteins.prot in 102s\n",
      "→ Processing 28446_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/28446_pfam.tbl\n",
      "✓ Finished 28446_predictedproteins.prot in 76s\n",
      "→ Processing 28109_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/28109_pfam.tbl\n",
      "✓ Finished 28109_predictedproteins.prot in 113s\n",
      "→ Processing 2300_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2300_pfam.tbl\n",
      "✓ Finished 2300_predictedproteins.prot in 186s\n",
      "→ Processing 2350_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/2350_pfam.tbl\n",
      "✓ Finished 2350_predictedproteins.prot in 172s\n",
      "→ Processing 28252_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/28252_pfam.tbl\n",
      "✓ Finished 28252_predictedproteins.prot in 133s\n",
      "→ Processing 262324_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/262324_pfam.tbl\n",
      "✓ Finished 262324_predictedproteins.prot in 198s\n",
      "→ Processing 290_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/290_pfam.tbl\n",
      "✓ Finished 290_predictedproteins.prot in 108s\n",
      "→ Processing 33018_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/33018_pfam.tbl\n",
      "✓ Finished 33018_predictedproteins.prot in 56s\n",
      "→ Processing 29323_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/29323_pfam.tbl\n",
      "✓ Finished 29323_predictedproteins.prot in 98s\n",
      "→ Processing 33977_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/33977_pfam.tbl\n",
      "✓ Finished 33977_predictedproteins.prot in 96s\n",
      "→ Processing 33952_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/33952_pfam.tbl\n",
      "✓ Finished 33952_predictedproteins.prot in 109s\n",
      "→ Processing 29355_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/29355_pfam.tbl\n",
      "✓ Finished 29355_predictedproteins.prot in 168s\n",
      "→ Processing 29370_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/29370_pfam.tbl\n",
      "✓ Finished 29370_predictedproteins.prot in 156s\n",
      "→ Processing 29369_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/29369_pfam.tbl\n",
      "✓ Finished 29369_predictedproteins.prot in 163s\n",
      "→ Processing 339_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/339_pfam.tbl\n",
      "✓ Finished 339_predictedproteins.prot in 165s\n",
      "→ Processing 39441_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/39441_pfam.tbl\n",
      "✓ Finished 39441_predictedproteins.prot in 80s\n",
      "→ Processing 36861_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/36861_pfam.tbl\n",
      "✓ Finished 36861_predictedproteins.prot in 113s\n",
      "→ Processing 34013_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/34013_pfam.tbl\n",
      "✓ Finished 34013_predictedproteins.prot in 194s\n",
      "→ Processing 38815_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/38815_pfam.tbl\n",
      "✓ Finished 38815_predictedproteins.prot in 107s\n",
      "→ Processing 35830_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/35830_pfam.tbl\n",
      "✓ Finished 35830_predictedproteins.prot in 145s\n",
      "→ Processing 354_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/354_pfam.tbl\n",
      "✓ Finished 354_predictedproteins.prot in 223s\n",
      "→ Processing 39777_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/39777_pfam.tbl\n",
      "✓ Finished 39777_predictedproteins.prot in 83s\n",
      "→ Processing 40542_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/40542_pfam.tbl\n",
      "✓ Finished 40542_predictedproteins.prot in 87s\n",
      "→ Processing 435_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/435_pfam.tbl\n",
      "✓ Finished 435_predictedproteins.prot in 77s\n",
      "→ Processing 41673_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/41673_pfam.tbl\n",
      "✓ Finished 41673_predictedproteins.prot in 88s\n",
      "→ Processing 510955_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/510955_pfam.tbl\n",
      "✓ Finished 510955_predictedproteins.prot in 45s\n",
      "→ Processing 45248_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/45248_pfam.tbl\n",
      "✓ Finished 45248_predictedproteins.prot in 133s\n",
      "→ Processing 50712_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/50712_pfam.tbl\n",
      "✓ Finished 50712_predictedproteins.prot in 133s\n",
      "→ Processing 53399_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/53399_pfam.tbl\n",
      "✓ Finished 53399_predictedproteins.prot in 88s\n",
      "→ Processing 533_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/533_pfam.tbl\n",
      "✓ Finished 533_predictedproteins.prot in 85s\n",
      "→ Processing 37331_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/37331_pfam.tbl\n",
      "✓ Finished 37331_predictedproteins.prot in 293s\n",
      "→ Processing 410_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/410_pfam.tbl\n",
      "✓ Finished 410_predictedproteins.prot in 209s\n",
      "→ Processing 542_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/542_pfam.tbl\n",
      "✓ Finished 542_predictedproteins.prot in 56s\n",
      "→ Processing 61595_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/61595_pfam.tbl\n",
      "✓ Finished 61595_predictedproteins.prot in 68s\n",
      "→ Processing 65555_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/65555_pfam.tbl\n",
      "✓ Finished 65555_predictedproteins.prot in 68s\n",
      "→ Processing 522889_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/522889_pfam.tbl\n",
      "✓ Finished 522889_predictedproteins.prot in 202s\n",
      "→ Processing 58138_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/58138_pfam.tbl\n",
      "✓ Finished 58138_predictedproteins.prot in 124s\n",
      "→ Processing 82803_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/82803_pfam.tbl\n",
      "✓ Finished 82803_predictedproteins.prot in 64s\n",
      "→ Processing 844_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/844_pfam.tbl\n",
      "✓ Finished 844_predictedproteins.prot in 39s\n",
      "→ Processing 691_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/691_pfam.tbl\n",
      "✓ Finished 691_predictedproteins.prot in 122s\n",
      "→ Processing 78346_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/78346_pfam.tbl\n",
      "✓ Finished 78346_predictedproteins.prot in 84s\n",
      "→ Processing 867_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/867_pfam.tbl\n",
      "✓ Finished 867_predictedproteins.prot in 62s\n",
      "→ Processing 76588_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/76588_pfam.tbl\n",
      "✓ Finished 76588_predictedproteins.prot in 156s\n",
      "→ Processing 876_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/876_pfam.tbl\n",
      "✓ Finished 876_predictedproteins.prot in 86s\n",
      "→ Processing 65655_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/65655_pfam.tbl\n",
      "✓ Finished 65655_predictedproteins.prot in 223s\n",
      "→ Processing 52697_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/52697_pfam.tbl\n",
      "✓ Finished 52697_predictedproteins.prot in 375s\n",
      "→ Processing 881_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/881_pfam.tbl\n",
      "✓ Finished 881_predictedproteins.prot in 117s\n",
      "→ Processing 879_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/879_pfam.tbl\n",
      "✓ Finished 879_predictedproteins.prot in 129s\n",
      "→ Processing 920_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/920_pfam.tbl\n",
      "✓ Finished 920_predictedproteins.prot in 75s\n",
      "→ Processing 891_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/891_pfam.tbl\n",
      "✓ Finished 891_predictedproteins.prot in 113s\n",
      "→ Processing 927_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/927_pfam.tbl\n",
      "✓ Finished 927_predictedproteins.prot in 53s\n",
      "→ Processing 894_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/894_pfam.tbl\n",
      "✓ Finished 894_predictedproteins.prot in 108s\n",
      "→ Processing 930_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/930_pfam.tbl\n",
      "✓ Finished 930_predictedproteins.prot in 61s\n",
      "→ Processing 54_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/54_pfam.tbl\n",
      "✓ Finished 54_predictedproteins.prot in 395s\n",
      "→ Processing 934_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/934_pfam.tbl\n",
      "✓ Finished 934_predictedproteins.prot in 64s\n",
      "→ Processing 931_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/931_pfam.tbl\n",
      "✓ Finished 931_predictedproteins.prot in 104s\n",
      "→ Processing 985_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/985_pfam.tbl\n",
      "✓ Finished 985_predictedproteins.prot in 61s\n",
      "→ Processing 971_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/971_pfam.tbl\n",
      "✓ Finished 971_predictedproteins.prot in 78s\n",
      "→ Processing 979_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/979_pfam.tbl\n",
      "✓ Finished 979_predictedproteins.prot in 81s\n",
      "→ Processing 921_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/921_pfam.tbl\n",
      "✓ Finished 921_predictedproteins.prot in 164s\n",
      "→ Processing 983_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/983_pfam.tbl\n",
      "✓ Finished 983_predictedproteins.prot in 79s\n",
      "✔ All done ( Tue May 20 21:57:01 BST 2025 ). Full log: /Users/liangxuntan/Code/fyp2025/data/logs/prot2pfam_20250520_212717.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['/Users/liangxuntan/Code/fyp2025/scripts/prot2pfam.sh', '-j', '8', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1022_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/106590_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1071_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1074_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1085_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1092_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1098_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1108_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/122960_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1247_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1267_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1280_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1313_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1314_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1404_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1445_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1461_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1497_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1498_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1511_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1515_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1519_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1520_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/152335_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1525_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1528_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1534_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/153721_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1553_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1556_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/163_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1673_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1744_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1771_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1792_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1833_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1852_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1877_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/209080_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2187_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2208_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2242_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2250_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2271_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2285_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2287_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2300_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2303_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2325_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/2350_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/238013_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/262324_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/271_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/274_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28109_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28252_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/28446_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/290_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29323_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29355_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29369_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/29370_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33952_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33977_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/339_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/34013_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/35830_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/36861_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/37331_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/38815_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39441_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/39777_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/40542_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/410_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/41673_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/435_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/45248_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/50712_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/510955_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/522889_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/52697_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/53399_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/533_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/542_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/58138_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/61595_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65555_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/65655_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/691_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/76588_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/78346_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/82803_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/844_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/867_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/876_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/879_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/881_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/891_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/894_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/920_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/921_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/927_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/930_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/931_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/934_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/971_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/979_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/983_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/985_predictedproteins.prot'], returncode=0)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gmc.run_prot2pfam(\"/Users/liangxuntan/Code/fyp2025/data/prodigal_output\",jobs=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8ff881e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(115, 2801)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Guanylate_cyc</th>\n",
       "      <th>HlyD_D23</th>\n",
       "      <th>SpoIIE</th>\n",
       "      <th>ETR1_N</th>\n",
       "      <th>DUF3422</th>\n",
       "      <th>Aminotran_1_2</th>\n",
       "      <th>AAA_21</th>\n",
       "      <th>AAA_15</th>\n",
       "      <th>Oxidored_nitro</th>\n",
       "      <th>tRNA-synt_1c</th>\n",
       "      <th>...</th>\n",
       "      <th>Zn_protease</th>\n",
       "      <th>NQRA_2nd</th>\n",
       "      <th>NQRA_N</th>\n",
       "      <th>Glyco_hydro_130</th>\n",
       "      <th>NTP_transf_4</th>\n",
       "      <th>Anticodon_3</th>\n",
       "      <th>DUF7834</th>\n",
       "      <th>PorZ_N_b_propeller</th>\n",
       "      <th>HMBD</th>\n",
       "      <th>RecG_dom3_C</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 2801 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Guanylate_cyc  HlyD_D23  SpoIIE  ETR1_N  DUF3422  Aminotran_1_2  AAA_21  \\\n",
       "155              1         3       2       1        2              1       1   \n",
       "159              1         0       0       0        0              0       0   \n",
       "37               1         0       0       0        0              1       0   \n",
       "166              1         1       0       0        0              2       0   \n",
       "157              1         0       0       0        0              0       0   \n",
       "\n",
       "     AAA_15  Oxidored_nitro  tRNA-synt_1c  ...  Zn_protease  NQRA_2nd  NQRA_N  \\\n",
       "155       3               4             2  ...            0         0       0   \n",
       "159       0               1             0  ...            0         0       0   \n",
       "37        0               0             2  ...            0         0       0   \n",
       "166       0               0             2  ...            0         0       0   \n",
       "157       0               0             0  ...            0         0       0   \n",
       "\n",
       "     Glyco_hydro_130  NTP_transf_4  Anticodon_3  DUF7834  PorZ_N_b_propeller  \\\n",
       "155                0             0            0        0                   0   \n",
       "159                0             0            0        0                   0   \n",
       "37                 0             0            0        0                   0   \n",
       "166                0             0            0        0                   0   \n",
       "157                0             0            0        0                   0   \n",
       "\n",
       "     HMBD  RecG_dom3_C  \n",
       "155     0            0  \n",
       "159     0            0  \n",
       "37      0            0  \n",
       "166     0            0  \n",
       "157     0            0  \n",
       "\n",
       "[5 rows x 2801 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam_df=gmc.build_pfam_matrix(\n",
    "    hmmer_dir_path=\"/Users/liangxuntan/Code/fyp2025/data/hmmer_output\",\n",
    "    taxid2media_dict=taxid_medianum_dict,\n",
    "    map_taxid=True,\n",
    "    save_path=\"/Users/liangxuntan/Code/fyp2025/data/dfs_output/pfam_matrix.csv\"\n",
    ")\n",
    "\n",
    "pfam_df = pfam_df.drop(index=77)\n",
    "print(pfam_df.shape)\n",
    "pfam_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "14ff35ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error 404 for medium ID 77\n",
      "Data saved to /Users/liangxuntan/Code/fyp2025/data/dfs_output/media_matrix.csv\n",
      "(115, 206)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>(DL)-alpha-Lipoic acid</th>\n",
       "      <th>(NH4) citrate</th>\n",
       "      <th>(NH4)2HPO4</th>\n",
       "      <th>(NH4)2SO4</th>\n",
       "      <th>(NH4)6Mo7O24 x 4 H2O</th>\n",
       "      <th>Agar</th>\n",
       "      <th>AlK(SO4)2 x 12 H2O</th>\n",
       "      <th>Allantoin</th>\n",
       "      <th>Ammonium acetate</th>\n",
       "      <th>Ammonium chloride</th>\n",
       "      <th>...</th>\n",
       "      <th>Vitamin K1</th>\n",
       "      <th>Vitamin K3</th>\n",
       "      <th>Water</th>\n",
       "      <th>Yeast extract</th>\n",
       "      <th>ZnCl2</th>\n",
       "      <th>ZnSO4 x 6 H2O</th>\n",
       "      <th>ZnSO4 x 7 H2O</th>\n",
       "      <th>air-dried garden soil</th>\n",
       "      <th>n-Valeric acid</th>\n",
       "      <th>p-Aminobenzoic acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 206 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     (DL)-alpha-Lipoic acid  (NH4) citrate  (NH4)2HPO4  (NH4)2SO4  \\\n",
       "155                       0              0           0          0   \n",
       "159                       0              0           0          1   \n",
       "37                        0              0           0          1   \n",
       "166                       0              0           0          1   \n",
       "157                       0              0           0          1   \n",
       "\n",
       "     (NH4)6Mo7O24 x 4 H2O  Agar  AlK(SO4)2 x 12 H2O  Allantoin  \\\n",
       "155                     0     1                   0          0   \n",
       "159                     0     0                   0          0   \n",
       "37                      0     0                   0          0   \n",
       "166                     1     1                   0          0   \n",
       "157                     0     0                   0          0   \n",
       "\n",
       "     Ammonium acetate  Ammonium chloride  ...  Vitamin K1  Vitamin K3  Water  \\\n",
       "155                 0                  0  ...           0           0      0   \n",
       "159                 0                  0  ...           0           0      0   \n",
       "37                  0                  0  ...           0           0      0   \n",
       "166                 0                  0  ...           0           0      0   \n",
       "157                 0                  0  ...           0           0      0   \n",
       "\n",
       "     Yeast extract  ZnCl2  ZnSO4 x 6 H2O  ZnSO4 x 7 H2O  \\\n",
       "155              0      0              0              1   \n",
       "159              0      0              0              0   \n",
       "37               0      0              0              0   \n",
       "166              0      0              0              1   \n",
       "157              1      0              0              0   \n",
       "\n",
       "     air-dried garden soil  n-Valeric acid  p-Aminobenzoic acid  \n",
       "155                      0               0                    0  \n",
       "159                      0               0                    0  \n",
       "37                       0               0                    0  \n",
       "166                      0               0                    0  \n",
       "157                      0               0                    0  \n",
       "\n",
       "[5 rows x 206 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "medianums= pfam_df.index.to_list()\n",
    "\n",
    "media_df=gmc.get_media_ingredient_matrix(medianums,\n",
    "                                         binary=True,\n",
    "                                         save_path=\"/Users/liangxuntan/Code/fyp2025/data/dfs_output/media_matrix.csv\")\n",
    "\n",
    "print(media_df.shape)\n",
    "media_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "35bdd25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: stratify=True requested but label counts too sparse; falling back to no stratification.\n",
      "\n",
      "Label: (DL)-alpha-Lipoic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: (NH4) citrate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: (NH4)2HPO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: (NH4)2SO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: (NH4)6Mo7O24 x 4 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Agar\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.722     0.929     0.812        14\n",
      "           1      0.800     0.444     0.571         9\n",
      "\n",
      "    accuracy                          0.739        23\n",
      "   macro avg      0.761     0.687     0.692        23\n",
      "weighted avg      0.753     0.739     0.718        23\n",
      "\n",
      "\n",
      "Label: AlK(SO4)2 x 12 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Allantoin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Ammonium acetate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Ammonium chloride\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Ascorbic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Bacto Soytone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Bacto peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Baker's yeast\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Beef extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Biotin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: Brain heart infusion\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Bromocresol purple\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Bromothymol blue\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Ca(NO3)2 x 4 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: CaCO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: CaCl2\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: CaCl2 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.857     0.353     0.500        17\n",
      "           1      0.312     0.833     0.455         6\n",
      "\n",
      "    accuracy                          0.478        23\n",
      "   macro avg      0.585     0.593     0.477        23\n",
      "weighted avg      0.715     0.478     0.488        23\n",
      "\n",
      "\n",
      "Label: CaCl2 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: CaSO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: CaSO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Calcium D-(+)-pantothenate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Calcium pantothenate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: Casamino acids\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Casein peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Casitone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Catalase\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Cellobiose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Cellulose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Co(NO3)2\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: CoCl2 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: CoSO4 x 7 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Crotonic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: CuCl2\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: CuCl2 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.739     1.000     0.850        17\n",
      "           1      0.000     0.000     0.000         6\n",
      "\n",
      "    accuracy                          0.739        23\n",
      "   macro avg      0.370     0.500     0.425        23\n",
      "weighted avg      0.546     0.739     0.628        23\n",
      "\n",
      "\n",
      "Label: CuSO4 x 5 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: CuSO4 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: D-(+)-biotin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: D-Fructose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: D-Glucose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: DL-2-Methylbutyric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: DL-mevalonic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Dextrose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Diaminopimelic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Disodium succinate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Distilled water\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: EDTA\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Ethanol\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Ethanol absolute\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Fe(III) citrate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Fe(III)PO4 x 4 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Fe(NH4)2(SO4)2\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Fe(NH4)2(SO4)2 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: FeCl2 x 4 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: FeCl3 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: FeSO4 x 7 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.682     0.938     0.789        16\n",
      "           1      0.000     0.000     0.000         7\n",
      "\n",
      "    accuracy                          0.652        23\n",
      "   macro avg      0.341     0.469     0.395        23\n",
      "weighted avg      0.474     0.652     0.549        23\n",
      "\n",
      "\n",
      "Label: Ferric ammonium citrate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Folic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Fructose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Fumaric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Glucose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Glutamic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Glycerol\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Glycyl-glycine\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Ground beef\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: H2SO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: H3BO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.609     1.000     0.757        14\n",
      "           1      0.000     0.000     0.000         9\n",
      "\n",
      "    accuracy                          0.609        23\n",
      "   macro avg      0.304     0.500     0.378        23\n",
      "weighted avg      0.371     0.609     0.461        23\n",
      "\n",
      "\n",
      "Label: HCl\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: Haemin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Hemin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Hexachlorocyclo-hexane\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Isobutyric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Isovaleric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: K-acetate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: K2CO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: K2HPO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.619     0.929     0.743        14\n",
      "           1      0.500     0.111     0.182         9\n",
      "\n",
      "    accuracy                          0.609        23\n",
      "   macro avg      0.560     0.520     0.462        23\n",
      "weighted avg      0.572     0.609     0.523        23\n",
      "\n",
      "\n",
      "Label: K2HPO4 x 3 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: KBr\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: KCl\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.696     1.000     0.821        16\n",
      "           1      0.000     0.000     0.000         7\n",
      "\n",
      "    accuracy                          0.696        23\n",
      "   macro avg      0.348     0.500     0.410        23\n",
      "weighted avg      0.484     0.696     0.571        23\n",
      "\n",
      "\n",
      "Label: KH2PO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.900     0.643     0.750        14\n",
      "           1      0.615     0.889     0.727         9\n",
      "\n",
      "    accuracy                          0.739        23\n",
      "   macro avg      0.758     0.766     0.739        23\n",
      "weighted avg      0.789     0.739     0.741        23\n",
      "\n",
      "\n",
      "Label: KNO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: KOH\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: L-Alanine\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: L-Cysteine HCl\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: L-Cysteine HCl x H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: L-Cystine\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: L-Glutathione\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Malt extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Maltose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Mannitol\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Meat extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Methanol\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Methylamine hydrochloride\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: MgCl2 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.696     1.000     0.821        16\n",
      "           1      0.000     0.000     0.000         7\n",
      "\n",
      "    accuracy                          0.696        23\n",
      "   macro avg      0.348     0.500     0.410        23\n",
      "weighted avg      0.484     0.696     0.571        23\n",
      "\n",
      "\n",
      "Label: MgSO4 x 7 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     0.182     0.308        11\n",
      "           1      0.571     1.000     0.727        12\n",
      "\n",
      "    accuracy                          0.609        23\n",
      "   macro avg      0.786     0.591     0.517        23\n",
      "weighted avg      0.776     0.609     0.527        23\n",
      "\n",
      "\n",
      "Label: MnCl2 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: MnCl2 x 4 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.739     1.000     0.850        17\n",
      "           1      0.000     0.000     0.000         6\n",
      "\n",
      "    accuracy                          0.739        23\n",
      "   macro avg      0.370     0.500     0.425        23\n",
      "weighted avg      0.546     0.739     0.628        23\n",
      "\n",
      "\n",
      "Label: MnSO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: MnSO4 x 4 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: MnSO4 x 5 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: MnSO4 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: MnSO4 x H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: MoO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: NH4Cl\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.696     1.000     0.821        16\n",
      "           1      0.000     0.000     0.000         7\n",
      "\n",
      "    accuracy                          0.696        23\n",
      "   macro avg      0.348     0.500     0.410        23\n",
      "weighted avg      0.484     0.696     0.571        23\n",
      "\n",
      "\n",
      "Label: NH4NO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Na glutamate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na-(DL)-lactate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-DL-lactate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-L-lactate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na-acetate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Na-acetate x 3 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-benzoate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-butyrate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na-formate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na-gluconate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-lactate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-propionate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na-thioglycolate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na2-DL-malate\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2-EDTA\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2-EDTA x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2-fumarate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na2-ß-glycerophosphate x 5 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2B4O7 x 10 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na2CO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: Na2CO3 anhydrous\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2HPO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2HPO4 x 12 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2HPO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Na2MoO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.652     1.000     0.789        15\n",
      "           1      0.000     0.000     0.000         8\n",
      "\n",
      "    accuracy                          0.652        23\n",
      "   macro avg      0.326     0.500     0.395        23\n",
      "weighted avg      0.425     0.652     0.515        23\n",
      "\n",
      "\n",
      "Label: Na2S x 9 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.652     1.000     0.789        15\n",
      "           1      0.000     0.000     0.000         8\n",
      "\n",
      "    accuracy                          0.652        23\n",
      "   macro avg      0.326     0.500     0.395        23\n",
      "weighted avg      0.425     0.652     0.515        23\n",
      "\n",
      "\n",
      "Label: Na2S2O3 x 5 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Na2SO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Na2SeO3 x 5 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Na2WO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Na3-citrate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: NaCl\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.652     1.000     0.789        15\n",
      "           1      0.000     0.000     0.000         8\n",
      "\n",
      "    accuracy                          0.652        23\n",
      "   macro avg      0.326     0.500     0.395        23\n",
      "weighted avg      0.425     0.652     0.515        23\n",
      "\n",
      "\n",
      "Label: NaF\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: NaH2PO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: NaH2PO4 x H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: NaHCO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: NaNO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: NaOH\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: NaSiO3 x 9 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: NaVO3 x H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Neopeptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: NiCl2 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.739     1.000     0.850        17\n",
      "           1      0.000     0.000     0.000         6\n",
      "\n",
      "    accuracy                          0.739        23\n",
      "   macro avg      0.370     0.500     0.425        23\n",
      "weighted avg      0.546     0.739     0.628        23\n",
      "\n",
      "\n",
      "Label: Nicotinic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Nitrilotriacetic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Nutrient broth\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Oat flakes\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: Phenol red\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Potassium phosphate buffer\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Proteose peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Proteose peptone no. 3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Putrescine\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Pyridoxine hydrochloride\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Pyruvic acid sodium salt\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Resazurin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Riboflavin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Rumen fluid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Skim milk\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Sludge\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Sodium Thiosulfate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Sodium malate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Sodium resazurin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.682     1.000     0.811        15\n",
      "           1      1.000     0.125     0.222         8\n",
      "\n",
      "    accuracy                          0.696        23\n",
      "   macro avg      0.841     0.562     0.517        23\n",
      "weighted avg      0.792     0.696     0.606        23\n",
      "\n",
      "\n",
      "Label: Sodium thiosulfate pentahydrate\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Soil extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Soy peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: SrCl2 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Starch\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Succinic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Sucrose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Sulfur\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Thiamine HCl\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.913     1.000     0.955        21\n",
      "           1      0.000     0.000     0.000         2\n",
      "\n",
      "    accuracy                          0.913        23\n",
      "   macro avg      0.457     0.500     0.477        23\n",
      "weighted avg      0.834     0.913     0.872        23\n",
      "\n",
      "\n",
      "Label: Thiamine-HCl x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: Tomato juice\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Tris-HCl buffer\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Trypticase\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Trypticase peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: Trypticase soy broth\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Tryptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.870     1.000     0.930        20\n",
      "           1      0.000     0.000     0.000         3\n",
      "\n",
      "    accuracy                          0.870        23\n",
      "   macro avg      0.435     0.500     0.465        23\n",
      "weighted avg      0.756     0.870     0.809        23\n",
      "\n",
      "\n",
      "Label: Tween 80\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Urea\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Uric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.957     1.000     0.978        22\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.957        23\n",
      "   macro avg      0.478     0.500     0.489        23\n",
      "weighted avg      0.915     0.957     0.935        23\n",
      "\n",
      "\n",
      "Label: VOSO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: VOSO4 x 5 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Valeric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Vitamin B12\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Label: Vitamin K1\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Vitamin K3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Water\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: Yeast extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     0.545     0.706        11\n",
      "           1      0.706     1.000     0.828        12\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.853     0.773     0.767        23\n",
      "weighted avg      0.847     0.783     0.769        23\n",
      "\n",
      "\n",
      "Label: ZnCl2\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: ZnSO4 x 6 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: ZnSO4 x 7 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.826     1.000     0.905        19\n",
      "           1      0.000     0.000     0.000         4\n",
      "\n",
      "    accuracy                          0.826        23\n",
      "   macro avg      0.413     0.500     0.452        23\n",
      "weighted avg      0.682     0.826     0.747        23\n",
      "\n",
      "\n",
      "Label: air-dried garden soil\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: n-Valeric acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000        23\n",
      "\n",
      "    accuracy                          1.000        23\n",
      "   macro avg      1.000     1.000     1.000        23\n",
      "weighted avg      1.000     1.000     1.000        23\n",
      "\n",
      "\n",
      "Label: p-Aminobenzoic acid\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.783     1.000     0.878        18\n",
      "           1      0.000     0.000     0.000         5\n",
      "\n",
      "    accuracy                          0.783        23\n",
      "   macro avg      0.391     0.500     0.439        23\n",
      "weighted avg      0.612     0.783     0.687        23\n",
      "\n",
      "\n",
      "Exact-match accuracy: 0.000\n",
      "Macro-averaged F1    : 0.018\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no true nor predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "clf, X_train, X_test, y_train, y_test, y_pred = mlf.train_and_evaluate_multioutput_rf(\n",
    "    pfam_df,\n",
    "    media_df,\n",
    "    stratify=True,\n",
    "    save_outputs=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1b9b5713",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Response_reg             0.008502\n",
      "tRNA-synt_2              0.008405\n",
      "MFS_1                    0.007693\n",
      "ShlB                     0.006656\n",
      "HD_5                     0.006575\n",
      "LytR_cpsA_psr            0.005561\n",
      "Transket_pyr             0.005506\n",
      "TonB_dep_Rec_b-barrel    0.005465\n",
      "AA_permease              0.005426\n",
      "AA_permease_2            0.005396\n",
      "dtype: float64\n"
     ]
    }
   ],
   "source": [
    "top10_pfams = mlf.get_top_feature_importances(clf, X_train, y_train, \"Glucose\", n=10)\n",
    "print(top10_pfams)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mne",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
