{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "90ab4983",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.append(\"/Users/liangxuntan/Code/fyp2025/scripts\")\n",
    "\n",
    "import gmcfunctions as gmc\n",
    "import mlfunctions as mlf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a85808b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error 404 for medium 4\n",
      "Error 404 for medium 5\n",
      "Error 404 for medium 15\n",
      "Error 404 for medium 16\n",
      "Error 404 for medium 17\n",
      "Error 404 for medium 18\n",
      "Error 404 for medium 19\n",
      "Error 404 for medium 20\n",
      "Error 404 for medium 22\n",
      "Error 404 for medium 23\n",
      "Error 404 for medium 24\n",
      "Error 404 for medium 30\n",
      "Error 404 for medium 33\n",
      "Error 404 for medium 41\n",
      "Error 404 for medium 42\n",
      "Error 404 for medium 44\n",
      "Error 404 for medium 45\n",
      "Error 404 for medium 47\n",
      "Error 404 for medium 50\n",
      "Error 404 for medium 62\n",
      "Error 404 for medium 64\n",
      "Error 404 for medium 66\n",
      "Error 404 for medium 72\n",
      "Error 404 for medium 77\n",
      "Error 404 for medium 89\n",
      "Error 404 for medium 94\n",
      "Error 404 for medium 96\n",
      "Error 404 for medium 99\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>...</th>\n",
       "      <th>86</th>\n",
       "      <th>87</th>\n",
       "      <th>88</th>\n",
       "      <th>90</th>\n",
       "      <th>91</th>\n",
       "      <th>92</th>\n",
       "      <th>93</th>\n",
       "      <th>95</th>\n",
       "      <th>97</th>\n",
       "      <th>98</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>(NH4) citrate</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>(NH4)2SO4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.28713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>(NH4)3 citrate</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>(NH4)6Mo7O24 x 4 H2O</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Agar</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   name     1     2     3     6     7     8     9    10   11  \\\n",
       "0         (NH4) citrate   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "1             (NH4)2SO4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "2        (NH4)3 citrate   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  2.0   \n",
       "3  (NH4)6Mo7O24 x 4 H2O   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "4                  Agar  15.0  15.0  15.0  15.0  15.0  15.0  15.0  15.0  NaN   \n",
       "\n",
       "   ...  86  87       88    90    91    92    93  95    97    98  \n",
       "0  ... NaN NaN      NaN   NaN   NaN   NaN   NaN NaN   NaN   NaN  \n",
       "1  ... NaN NaN  1.28713   NaN   NaN   NaN   NaN NaN   NaN   NaN  \n",
       "2  ... NaN NaN      NaN   NaN   NaN   NaN   NaN NaN   NaN   NaN  \n",
       "3  ... NaN NaN      NaN   NaN   NaN   NaN   NaN NaN   NaN   NaN  \n",
       "4  ... NaN NaN      NaN  15.0  15.0  15.0  15.0 NaN  20.0  15.0  \n",
       "\n",
       "[5 rows x 72 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "medianumlist= range(1, 10)\n",
    "media_df=gmc.fetch_medium_compositions(medianumlist)\n",
    "media_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4cf9d979",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error 404 for 4\n",
      "Error 404 for 5\n",
      "{1: 654.0, 2: 11979, 3: 12751, 4: None, 5: None, 6: 665, 7: 10435, 8: 14385, 9: 10177.0}\n"
     ]
    }
   ],
   "source": [
    "medianum_bacid_dict=gmc.get_first_bacdive_ids(medianumlist)\n",
    "print(medianum_bacid_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e0885d18",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Authentication successful --\n",
      "medium 1: BacDive ID → 654.0\n",
      "  » species-level taxid 1398\n",
      "medium 2: BacDive ID → 11979\n",
      "  » species-level taxid 1474\n",
      "medium 3: BacDive ID → 12751\n",
      "  » species-level taxid 354\n",
      "medium 4: BacDive ID → None\n",
      "No valid BacDive ID for medium 4\n",
      "medium 5: BacDive ID → None\n",
      "No valid BacDive ID for medium 5\n",
      "medium 6: BacDive ID → 665\n",
      "  » species-level taxid 1458\n",
      "medium 7: BacDive ID → 10435\n",
      "  » species-level taxid 190\n",
      "medium 8: BacDive ID → 14385\n",
      "  » species-level taxid 33018\n",
      "medium 9: BacDive ID → 10177.0\n",
      "  » species-level taxid 54\n",
      "{1398: 1, 1474: 2, 354: 3, 1458: 6, 190: 7, 33018: 8, 54: 9}\n"
     ]
    }
   ],
   "source": [
    "taxid_medianum_dict=gmc.map_species_taxids_to_media(\n",
    "                    medianums=medianumlist,\n",
    "                    bacdive_id_dict=medianum_bacid_dict,\n",
    "                    email=\"tanliangxun2000@gmail.com\",\n",
    "                    password=\"pass1234\",)\n",
    "print(taxid_medianum_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "41c688b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: /Users/liangxuntan/Code/fyp2025/scripts/run_taxid2fna.sh 1398 1474 354 1458 190 33018 54\n",
      "→ Processing TaxID 1398\n",
      "→ Processing TaxID 1474\n",
      "→ Processing TaxID 354\n",
      "→ Processing TaxID 1458\n",
      "→ Processing TaxID 190\n",
      "→ Processing TaxID 33018\n",
      "→ Processing TaxID 54\n",
      "✓ Finished. Full log: /Users/liangxuntan/Code/fyp2025/data/logs/taxid2fna_20250519_223336.log\n"
     ]
    }
   ],
   "source": [
    "taxids= list(taxid_medianum_dict.keys())\n",
    "\n",
    "gmc.run_taxid2fna(taxids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b14cf339",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 166 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 121 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot'\n",
      "Written 69 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot'\n",
      "Written 103 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 493 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Written 58 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot'\n",
      "Written 276 protein sequences to '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot'\n",
      "Protein prediction output files:\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot\n",
      "/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot'),\n",
       " PosixPath('/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot')]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gmc.run_protein_prediction(\n",
    "    fasta_dir_path=\"/Users/liangxuntan/Code/fyp2025/data/labelled_genomes\",\n",
    "    evalue='1e-20',\n",
    "    nproc=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e09f0675",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: /Users/liangxuntan/Code/fyp2025/scripts/prot2pfam.sh -j 4 /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot /Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot\n",
      "▶ Starting parallel hmmscan ( Mon May 19 23:16:55 BST 2025 ) on 7 file(s)\n",
      "Running 4 job(s) × 1 thread(s) each\n",
      "→ Processing 1398_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1398_pfam.tbl\n",
      "✓ Finished 1398_predictedproteins.prot in 20s\n",
      "→ Processing 190_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/190_pfam.tbl\n",
      "✓ Finished 190_predictedproteins.prot in 34s\n",
      "→ Processing 1474_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1474_pfam.tbl\n",
      "✓ Finished 1474_predictedproteins.prot in 40s\n",
      "→ Processing 33018_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/33018_pfam.tbl\n",
      "✓ Finished 33018_predictedproteins.prot in 22s\n",
      "→ Processing 1458_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/1458_pfam.tbl\n",
      "✓ Finished 1458_predictedproteins.prot in 52s\n",
      "→ Processing 354_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/354_pfam.tbl\n",
      "✓ Finished 354_predictedproteins.prot in 64s\n",
      "→ Processing 54_predictedproteins.prot ➜ /Users/liangxuntan/Code/fyp2025/data/hmmer_output/54_pfam.tbl\n",
      "✓ Finished 54_predictedproteins.prot in 97s\n",
      "✔ All done ( Mon May 19 23:19:12 BST 2025 ). Full log: /Users/liangxuntan/Code/fyp2025/data/logs/prot2pfam_20250519_231655.log\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['/Users/liangxuntan/Code/fyp2025/scripts/prot2pfam.sh', '-j', '4', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1398_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1458_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/1474_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/190_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/33018_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/354_predictedproteins.prot', '/Users/liangxuntan/Code/fyp2025/data/prodigal_output/54_predictedproteins.prot'], returncode=0)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gmc.run_prot2pfam(\"/Users/liangxuntan/Code/fyp2025/data/prodigal_output\",jobs=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8ff881e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7, 718)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>COX1</th>\n",
       "      <th>Mur_ligase_M</th>\n",
       "      <th>DEAD</th>\n",
       "      <th>Helicase_C</th>\n",
       "      <th>ATP-grasp_2</th>\n",
       "      <th>Ligase_CoA</th>\n",
       "      <th>RNase_J_b_CASP</th>\n",
       "      <th>RNase_J_C</th>\n",
       "      <th>Aldedh</th>\n",
       "      <th>AMP-binding</th>\n",
       "      <th>...</th>\n",
       "      <th>Trp_syntA</th>\n",
       "      <th>RlaP</th>\n",
       "      <th>CoA_transf_3</th>\n",
       "      <th>ABC2_membrane_6</th>\n",
       "      <th>UDPGP</th>\n",
       "      <th>MR_MLE_C</th>\n",
       "      <th>YARHG</th>\n",
       "      <th>ParBc</th>\n",
       "      <th>ACP_syn_III_C</th>\n",
       "      <th>ACP_syn_III</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 718 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   COX1  Mur_ligase_M  DEAD  Helicase_C  ATP-grasp_2  Ligase_CoA  \\\n",
       "1     2             1     2           1            1           1   \n",
       "6     0             2     1           2            0           0   \n",
       "7     0             2     2           1            1           1   \n",
       "3     0             2     0           0            0           1   \n",
       "8     0             0     3           2            0           0   \n",
       "\n",
       "   RNase_J_b_CASP  RNase_J_C  Aldedh  AMP-binding  ...  Trp_syntA  RlaP  \\\n",
       "1               1          1       6            1  ...          0     0   \n",
       "6               0          0       0            0  ...          0     0   \n",
       "7               0          0       2            0  ...          0     0   \n",
       "3               0          0       0            1  ...          0     0   \n",
       "8               0          0       6            0  ...          0     0   \n",
       "\n",
       "   CoA_transf_3  ABC2_membrane_6  UDPGP  MR_MLE_C  YARHG  ParBc  \\\n",
       "1             0                0      0         0      0      0   \n",
       "6             0                0      0         0      0      0   \n",
       "7             0                0      0         0      0      0   \n",
       "3             0                0      0         0      0      0   \n",
       "8             0                0      0         0      0      0   \n",
       "\n",
       "   ACP_syn_III_C  ACP_syn_III  \n",
       "1              0            0  \n",
       "6              0            0  \n",
       "7              0            0  \n",
       "3              0            0  \n",
       "8              0            0  \n",
       "\n",
       "[5 rows x 718 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam_df=gmc.build_pfam_matrix(\n",
    "    hmmer_dir_path=\"/Users/liangxuntan/Code/fyp2025/data/hmmer_output\",\n",
    "    taxid2media_dict=taxid_medianum_dict,\n",
    "    map_taxid=True,\n",
    "    save_path=\"/Users/liangxuntan/Code/fyp2025/data/dfs_output/pfam_matrix.csv\"\n",
    ")\n",
    "\n",
    "print(pfam_df.shape)\n",
    "pfam_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "14ff35ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data saved to /Users/liangxuntan/Code/fyp2025/data/dfs_output/media_matrix.csv\n",
      "(7, 19)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Agar</th>\n",
       "      <th>Allantoin</th>\n",
       "      <th>Baker's yeast</th>\n",
       "      <th>CaCO3</th>\n",
       "      <th>CaCl2 x 2 H2O</th>\n",
       "      <th>Distilled water</th>\n",
       "      <th>FeSO4 x 7 H2O</th>\n",
       "      <th>Glucose</th>\n",
       "      <th>K2HPO4</th>\n",
       "      <th>KH2PO4</th>\n",
       "      <th>Mannitol</th>\n",
       "      <th>Meat extract</th>\n",
       "      <th>MgSO4 x 7 H2O</th>\n",
       "      <th>MnSO4 x H2O</th>\n",
       "      <th>Na2MoO4 x 2 H2O</th>\n",
       "      <th>Peptone</th>\n",
       "      <th>Urea</th>\n",
       "      <th>Vitamin B12</th>\n",
       "      <th>Yeast extract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Agar  Allantoin  Baker's yeast  CaCO3  CaCl2 x 2 H2O  Distilled water  \\\n",
       "1     1          0              0      0              0                0   \n",
       "6     1          1              0      0              1                0   \n",
       "7     1          0              0      0              0                0   \n",
       "3     1          0              0      1              1                0   \n",
       "8     1          0              0      1              0                0   \n",
       "\n",
       "   FeSO4 x 7 H2O  Glucose  K2HPO4  KH2PO4  Mannitol  Meat extract  \\\n",
       "1              0        0       0       0         0             1   \n",
       "6              1        0       1       1         0             0   \n",
       "7              0        1       0       0         0             0   \n",
       "3              1        1       1       1         1             0   \n",
       "8              0        1       0       0         0             0   \n",
       "\n",
       "   MgSO4 x 7 H2O  MnSO4 x H2O  Na2MoO4 x 2 H2O  Peptone  Urea  Vitamin B12  \\\n",
       "1              0            0                0        1     0            0   \n",
       "6              1            1                0        0     0            0   \n",
       "7              0            0                0        1     0            0   \n",
       "3              1            0                1        0     0            0   \n",
       "8              0            0                0        1     0            0   \n",
       "\n",
       "   Yeast extract  \n",
       "1              0  \n",
       "6              0  \n",
       "7              1  \n",
       "3              0  \n",
       "8              1  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "medianums= pfam_df.index.to_list()\n",
    "\n",
    "media_df=gmc.get_media_ingredient_matrix(medianums,\n",
    "                                         binary=True,\n",
    "                                         save_path=\"/Users/liangxuntan/Code/fyp2025/data/dfs_output/media_matrix.csv\")\n",
    "\n",
    "print(media_df.shape)\n",
    "media_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "35bdd25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: stratify=True requested but label counts too sparse; falling back to no stratification.\n",
      "\n",
      "Label: Agar\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           1      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: Allantoin\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: Baker's yeast\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: CaCO3\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: CaCl2 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: Distilled water\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: FeSO4 x 7 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: Glucose\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.000     0.000     0.000       2.0\n",
      "           1      0.000     0.000     0.000       0.0\n",
      "\n",
      "    accuracy                          0.000       2.0\n",
      "   macro avg      0.000     0.000     0.000       2.0\n",
      "weighted avg      0.000     0.000     0.000       2.0\n",
      "\n",
      "\n",
      "Label: K2HPO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: KH2PO4\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: Mannitol\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: Meat extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: MgSO4 x 7 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: MnSO4 x H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.500     1.000     0.667         1\n",
      "           1      0.000     0.000     0.000         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: Na2MoO4 x 2 H2O\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: Peptone\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      0.000     0.000     0.000         1\n",
      "           1      0.500     1.000     0.667         1\n",
      "\n",
      "    accuracy                          0.500         2\n",
      "   macro avg      0.250     0.500     0.333         2\n",
      "weighted avg      0.250     0.500     0.333         2\n",
      "\n",
      "\n",
      "Label: Urea\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: Vitamin B12\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Label: Yeast extract\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0      1.000     1.000     1.000         2\n",
      "\n",
      "    accuracy                          1.000         2\n",
      "   macro avg      1.000     1.000     1.000         2\n",
      "weighted avg      1.000     1.000     1.000         2\n",
      "\n",
      "\n",
      "Exact-match accuracy: 0.000\n",
      "Macro-averaged F1    : 0.088\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Recall is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Recall is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Recall is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Users/liangxuntan/anaconda3/envs/mne/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1531: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no true nor predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "clf, X_train, X_test, y_train, y_test, y_pred = mlf.train_and_evaluate_multioutput_rf(\n",
    "    pfam_df,\n",
    "    media_df,\n",
    "    stratify=True,\n",
    "    save_outputs=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1b9b5713",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bac_luciferase     0.023965\n",
      "Peptidase_M28      0.017429\n",
      "AIRS               0.016340\n",
      "UvrC_RNaseH_dom    0.016340\n",
      "Lactamase_B        0.014161\n",
      "MFS_1              0.013072\n",
      "Bac_DnaA           0.011983\n",
      "TelA               0.010893\n",
      "Sigma54_activat    0.010893\n",
      "Aminotran_3        0.009804\n",
      "dtype: float64\n"
     ]
    }
   ],
   "source": [
    "top10_pfams = mlf.get_top_feature_importances(clf, X_train, y_train, \"Glucose\", n=10)\n",
    "print(top10_pfams)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mne",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
